gopherage/pkg/cov/junit/calculation/coveragelist.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package calculation
+
+import (
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+)
+
+// CoverageList is a collection and summary over multiple file Coverage objects
+type CoverageList struct {
+	*coverage
+	Group            []coverage
+	summaryCompleted bool
+}
+
+// CovList constructs new (file) Group Coverage
+func newCoverageList(name string) *CoverageList {
+	return &CoverageList{
+		coverage:         &coverage{Name: name},
+		Group:            []coverage{},
+		summaryCompleted: false,
+	}
+}
+
+// Percentage returns the percentage of statements covered
+func (covList *CoverageList) Percentage() string {
+	return fmt.Sprintf("%.1f", covList.Ratio()*100)
+}
+
+//Ratio summarizes the list of coverages and returns the summarized ratio
+func (covList *CoverageList) Ratio() float32 {
+	covList.Summarize()
+	return covList.coverage.Ratio()
+}
+
+// Summarize summarizes all items in the Group and stores the result
+func (covList *CoverageList) Summarize() {
+	if !covList.summaryCompleted {
+		for _, item := range covList.Group {
+			covList.nCoveredStmts += item.nCoveredStmts
+			covList.nAllStmts += item.nAllStmts
+		}
+		covList.summaryCompleted = true
+	}
+}
+
+func (covList *CoverageList) append(c *coverage) {
+	covList.Group = append(covList.Group, *c)
+}
+
+// Subset returns the subset obtained through applying filter
+func (covList *CoverageList) Subset(prefix string) *CoverageList {
+	s := newCoverageList("Filtered Summary")
+	for _, c := range covList.Group {
+		if strings.HasPrefix(c.Name, prefix) {
+			s.append(&c)
+		}
+	}
+	return s
+}
+
+// ListDirectories gets a list a sub-directories that contains source code.
+func (covList CoverageList) ListDirectories() []string {
+	dirSet := map[string]bool{}
+	for _, cov := range covList.Group {
+		dirSet[path.Dir(cov.Name)] = true
+	}
+	var result []string
+	for key := range dirSet {
+		result = append(result, key)
+	}
+	sort.Strings(result)
```

Comment:
Do we need to sort?

Reply:
does testgrid do the sorting itself and ignores the order in the xml file?

Reply:
@srinivashegde86